crates/pallets/starknet/src/lib.rs

```diff
@@ -874,7 +874,13 @@ impl<T: Config> Pallet<T> {
         let sequencer_address = SEQUENCER_ADDRESS;
         let block_timestamp = Self::block_timestamp();
         let transaction_count = pending.len() as u128;
-        let transactions: Vec<Transaction> = pending.into_iter().map(|(transaction, _)| transaction).collect();
+
+        // TODO: need more experienced rust dev here to understand
+        // what is the correct way of doing the next two clones.
```

Comment:
What is the problem here exactly?

Did you tried using `iter` rather than `into_iter` and only cloning in the end?

Reply:
I have changed here to support the two fields `transactions` and `transaction_receipts` of the block. I have made one for loop to ensure we only iterate one time on `pending`. Once again, I didn't manage to get this with `iter` and `map`. Let me know in the new commit if it's ok. :+1: